ecommerce/api.py

```diff
@@ -96,8 +97,10 @@ def get_purchasable_course_run(course_key, user):
             order__user=user,
             course_key=course_run.edx_course_key,
```

Comment:
This Identifies if a user has paid for a particular course_run. For each paid course_run you have 2 exam attempts. So to schedule a third attempt you would need to pay.